compiler/luci/pass/include/luci/Pass/QuantizeDequantizeWeightsPass.h

```diff
@@ -31,23 +31,38 @@ namespace luci
  */
 class QuantizeDequantizeWeightsPass : public logo::Pass
 {
+public:
+  struct Context
+  {
+    loco::DataType input_model_dtype = loco::DataType::Unknown;
+    loco::DataType output_model_dtype = loco::DataType::Unknown;
+    QuantizationGranularity granularity = QuantizationGranularity::ChannelWise;
+  };
+
+public:
+  QuantizeDequantizeWeightsPass(std::unique_ptr<Context> &&ctx) : _ctx{std::move(ctx)}
+  {
+    // DO NOTHING
+  }
+
 public:
   QuantizeDequantizeWeightsPass(loco::DataType input_model_dtype, loco::DataType output_model_dtype,
                                 QuantizationGranularity granularity)
-    : _input_model_dtype{input_model_dtype}, _output_model_dtype{output_model_dtype}, _granularity{
-                                                                                        granularity}
   {
-    // DO NOTHING
+    _ctx = std::make_unique<Context>();
+    {
+      _ctx->input_model_dtype = input_model_dtype;
+      _ctx->output_model_dtype = output_model_dtype;
+      _ctx->granularity = granularity;
+    }
   }
```

Comment:
How about use initializer list?
```cxx
QuantizeDequantizeWeightsPass(loco::DataType input_model_dtype, loco::DataType output_model_dtype,
                                QuantizationGranularity granularity) : _ctx{std::make_unique<Context>(Context{input_model_dtype, output_model_dtype, granularity})}
{
  // DO NOTHING
}
```
Or
```cxx
// need constructor of `Context`
QuantizeDequantizeWeightsPass(loco::DataType input_model_dtype, loco::DataType output_model_dtype,
                                QuantizationGranularity granularity) : _ctx{std::make_unique<Context>(input_model_dtype, output_model_dtype, granularity)}
{
  // DO NOTHING
}
```

Reply:
I like @jinevening 's code as it's easy to read :)

Reply:
In addition to @seanshpark 's comment, I'll add a new variable to `Context`, which may change the constructor signature. So, I'd like to keep the current style :)